SignalMessaging/Views/ContactCellView.m

```diff
@@ -0,0 +1,309 @@
+
+@property (nonatomic) UILabel *nameLabel;
+@property (nonatomic) UILabel *profileNameLabel;
+@property (nonatomic) UIImageView *avatarView;
+@property (nonatomic) UILabel *subtitleLabel;
+@property (nonatomic) UILabel *ows_accessoryLabel;
+@property (nonatomic) UIStackView *nameContainerView;
+@property (nonatomic) UIView *ows_accessoryViewContainer;
+
+@property (nonatomic) OWSContactsManager *contactsManager;
+@property (nonatomic) NSString *recipientId;
+
+@end
+
+#pragma mark -
+
+@implementation ContactCellView
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        [self configure];
+    }
+    return self;
+}
+
+- (void)configure
+{
+    OWSAssert(!self.nameLabel);
+
+    self.layoutMargins = UIEdgeInsetsZero;
+
+    _avatarView = [AvatarImageView new];
+    [_avatarView autoSetDimension:ALDimensionWidth toSize:kContactCellAvatarSize];
+    [_avatarView autoSetDimension:ALDimensionHeight toSize:kContactCellAvatarSize];
+
+    self.nameLabel = [UILabel new];
+    self.nameLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+    self.nameLabel.textColor = [UIColor blackColor];
+
+    self.profileNameLabel = [UILabel new];
+    self.profileNameLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+    self.profileNameLabel.textColor = [UIColor grayColor];
+
+    self.subtitleLabel = [UILabel new];
+    self.subtitleLabel.textColor = [UIColor ows_darkGrayColor];
+
+    self.ows_accessoryLabel = [[UILabel alloc] init];
+    self.ows_accessoryLabel.textAlignment = NSTextAlignmentRight;
+    self.ows_accessoryLabel.textColor = [UIColor colorWithWhite:0.5f alpha:1.f];
+
+    self.ows_accessoryViewContainer = [UIView containerView];
+
+    self.nameContainerView = [[UIStackView alloc] initWithArrangedSubviews:@[
+        self.nameLabel,
+        self.profileNameLabel,
+        self.subtitleLabel,
+    ]];
+    self.nameContainerView.axis = UILayoutConstraintAxisVertical;
+    self.nameContainerView.alignment = UIStackViewAlignmentFill;
+
+    [self.avatarView setContentHuggingHorizontalHigh];
+    [self.nameContainerView setContentHuggingHorizontalLow];
+    [self.ows_accessoryViewContainer setContentHuggingHorizontalHigh];
+
+    UIStackView *hStackView = [[UIStackView alloc] initWithArrangedSubviews:@[
+        self.avatarView,
+        self.nameContainerView,
+        self.ows_accessoryViewContainer,
+    ]];
+    hStackView.axis = UILayoutConstraintAxisHorizontal;
+    hStackView.spacing = kContactCellAvatarTextMargin;
+    hStackView.distribution = UIStackViewDistributionFill;
+    [self addSubview:hStackView];
+    [hStackView autoVCenterInSuperview];
+    [hStackView autoPinLeadingToSuperviewMargin];
+    [hStackView autoPinTrailingToSuperviewMargin];
+    // Ensure that the cell's contents never overflow the cell bounds.
+    [hStackView autoPinEdgeToSuperviewMargin:ALEdgeTop relation:NSLayoutRelationGreaterThanOrEqual];
+    [hStackView autoPinEdgeToSuperviewMargin:ALEdgeBottom relation:NSLayoutRelationGreaterThanOrEqual];
```

Comment:
Much of this logic is pulled over wholesale from `ContactTableViewCell`, but I took the opportunity to clean up. For example, I use stack views where possible.

Reply:
👍